libsql-sqlite3/src/main.c

```diff
@@ -3356,8 +3356,8 @@ static int openDatabase(
   ){
     db->mutex = sqlite3MutexAlloc(SQLITE_MUTEX_RECURSIVE);
     if( db->mutex==0 ){
-      wal_manager->ref.xDestroy(wal_manager->ref.pData);
-      sqlite3_free(db->wal_manager);
+      db->wal_manager->ref.xDestroy(db->wal_manager->ref.pData);
```

Comment:
Isn't db->wal_manager == wal_manager always?

Reply:
Yes - I just wanted to make access to `wal_manager` consistent here. 
Can change to `wal_manager` in the block if you want. Don't know which is better actually...

Reply:
No need for any changes. Just making sure I'm not missing anything.

In general, it's better to avoid changes that are not necessary because it such change might cause conflict next time we will be merging upstream sqlite into libsql.

Reply:
Yes, nice point.

But both these lines are added by `libsql` and `sqlite` source just need to free DB like this:
```c
db->mutex = sqlite3MutexAlloc(SQLITE_MUTEX_RECURSIVE);
if( db->mutex==0 ){
  sqlite3_free(db);
  db = 0;
  goto opendb_out;
}
```

So I think it's fine to write these lines as we want